qiskit/assembler/assemble_schedules.py

```diff
@@ -154,14 +157,15 @@ def _assemble_instructions(
         instruction_converter: A converter instance which can convert PulseInstructions to
                                PulseQobjInstructions.
         run_config: Configuration of the runtime environment.
+        user_pulselib: User pulse library from previous schedule.
 
     Returns:
         A list of converted instructions, the user pulse library dictionary (from pulse name to
         pulse command), and the maximum number of readout memory slots used by this Schedule.
     """
     max_memory_slot = 0
     qobj_instructions = []
-    user_pulselib = {}
+    user_pulselib = deepcopy(user_pulselib)
```

Comment:
Can this be avoided? Can get slow if there are many pulses. Would it work if this line was removed?

Reply:
I checked out locally, this line and line 108 can be removed, and `user_pulses` -> `user_pulse_lib` in line 103

Reply:
ok. I thought you don't prefer such mutability. I'll remove this.